icubam/www/handlers/update.py

```diff
@@ -27,47 +27,19 @@ def get_consent_html(self, user):
       with open(path, 'r') as fp:
         return fp.read()
 
-  def authenticate_from_token(self, user_token):
-    input_data = self.token_encoder.decode(user_token)
-    if input_data is None:
-      logging.error("No token to be found.")
-      return None, None
-
-    userid = int(input_data.get('user_id', -1))
-    user = self.db.get_user(userid)
-    if user is None:
-      logging.error(f"User does not exist.")
-      return None, None
-
-    icuid = int(input_data.get('icu_id', -1))
-    user_icu_ids = [i.icu_id for i in user.icus]
-    if icuid not in user_icu_ids:
-      logging.error(f"User does not belong the ICU.")
-      return None, None
-
-    if user.consent is not None and not user.consent:
-      logging.error(f"User has bailed out from ICUBAM.")
-      return None, None
-
-    return user, input_data
-
   async def get(self):
     """Serves the page with a form to be filled by the user."""
     user_token = self.get_query_argument(self.QUERY_ARG)
-    user, token_data = self.authenticate_from_token(user_token)
-    if user is None:
+    user, icu = self.decode_token(user_token)
```

Comment:
I just want to double-check that we're not making any auth sacrifices by getting rid of `authenticate_from_token`?

Reply:
no, actually i am removing duplicated code and moving authenticated_from_token to base.py while renaming it decode_token. It now returns the user and the icu objects.

This is better IMO since we use the same function here and in home.py (the map) instead of somewhat replicating the code.